modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/twostep/GridMapQueryExecutor.java

```diff
@@ -645,6 +676,8 @@ public void onDmlRequest(final ClusterNode node, final GridH2DmlRequest req) {
                 reserved.release();
 
             nodeResults.removeUpdate(reqId);
+
+            trace.close();
```

Comment:
Try with resources?

Reply:
Unfortunately I can't use try with resource here. From my point of view it's quite important to trace messages that are sent from the catch block too.